Upstream-Status: Pending

Author: Kai-Chung Yan <seamlikok@gmail.com>
Last-Update: 2016-08-24
Description: Manual definition of struct user_pt_regs
 On ARM64, libunwind uses struct user_pt_regs which is not defined in
 anywhere, which causes FTBFS.
Forwarded: not-needed
--- a/external/libunwind/src/ptrace/_UPT_access_reg.c
+++ b/external/libunwind/src/ptrace/_UPT_access_reg.c
@@ -26,6 +26,15 @@ WITH THE SOFTWARE OR THE USE OR OTHER DE
 
 #include "_UPT_internal.h"
 
+#if defined(__aarch64__)
+  struct user_pt_regs {
+    __u64		regs[31];
+    __u64		sp;
+    __u64		pc;
+    __u64		pstate;
+  };
+#endif
+
 #if UNW_TARGET_IA64
 # include <elf.h>
 # ifdef HAVE_ASM_PTRACE_OFFSETS_H
